classes/task/duplicate_task.php

```diff
@@ -49,7 +49,20 @@ class duplicate_task extends adhoc_task {
      */
     public function execute() {
         $data = $this->get_custom_data();
-        $sectionid = empty($data->sectionid) ? false : $data->sectionid;
-        actions::duplicate((array) $data->modules, $sectionid);
+        // We use sectionid for duplicating modules in the same course, but sectionnum for duplicating to another course.
```

Comment:
ideally we should probably use better names like "selfsectionid" and "targetsectionid" (not that those names are perfect, but maybe a bit less confusing)

Reply:
Both of these are *target* sections. The - I admit - a bit confusing thing is that I am using the section id (like the first column in the table `sections`) for duplicating in the same course, and the section *number* (starting from 0, 1, 2 in each course) when duplicating to a different course, just from how the functions for doing this work right now. Any suggestion on how to clarify that?

Reply:
Not without getting too verbose I guess.  The commenting makes it clear enough, but I found myself constantly trying to remember which was which.  I say leave it.  It was a very minor thing anyway.